prow/pod-utils/downwardapi/jobspec.go

```diff
@@ -123,17 +126,23 @@ func EnvForSpec(spec JobSpec) (map[string]string, error) {
 	if spec.Type == kube.PeriodicJob {
 		return env, nil
 	}
-	env[repoOwnerEnv] = spec.Refs.Org
-	env[repoNameEnv] = spec.Refs.Repo
-	env[pullBaseRefEnv] = spec.Refs.BaseRef
-	env[pullBaseShaEnv] = spec.Refs.BaseSHA
-	env[pullRefsEnv] = spec.Refs.String()
+
+	if len(spec.Refs) > 0 {
```

Comment:
Not super happy about this

Reply:
There are a few postsubmit/batch test cases didn't specify refs... I can fix that instead

Reply:
Yeah, only periodic should be allowed not to have refs

Reply:
fixed in tests